src/lib/dataset/mongo.ts

```diff
@@ -67,3 +67,38 @@ export function _guessType(val: any, prevType: any = null): DataSetColumnType |
 
   return null;
 }
+
+/**
+ * extend a `DataSet` headers with inferred types, base on value of columns
+ */
+export function inferTypeFromDataset(dataset: DataSet, maxRows: number = 50): DataSet {
+  // For the moment, we infer arbitrarily column's types on maximum the 50 first rows
+  maxRows = Math.min(dataset.data.length, maxRows);
+  const newHeaders: DataSetColumn[] = [];
+
+  dataset.headers.forEach((header, i) => {
```

Comment:
I would use of `for..of` loop on `header.entries()` although you might need to set `downlevelIteration` in `tscfongig.json` to be perform a `for of` loop on an iterator. Otherwise, `Object.entries()` should also do the trick:

```typescript
for (const [i, header] of dataset.headers.entries()) {
   // ...
} 
```

Also, it might be worth to rename `i` into `colidx` :-)